[OpenAPIGenerator] Add section about issues during `anyOf/oneOf` (#524)
SAP/cloud-sdk: docs/java/features/rest/generate-rest-client.mdx

</TabItem>
</Tabs>

## Solving Compilation Issues After Generation

:::info

From version `3.58` of the SAP Cloud SDK, the OpenAPI generator has been enhanced to fail if the input specification contains keywords `anyOf` or `oneOf` in certain positions.

> (More details on these positions below)

Reviewer: ```suggestion
```

<details>
  <summary>Keywords under properties of schema</summary>

```
components:
  schemas:
    SampleSchema:
      title: SampleSchema
      description: ...
      type: object
      properties:
        embeddedData:
          $ref: '#/components/schemas/EmbeddedData'
        questions:
          type: array
          title: questions
          items:
            title: Question
            oneOf:
              - $ref: '#/components/schemas/mc'
              - $ref: '#/components/schemas/te'
              - $ref: '#/components/schemas/db'
            discriminator:
              propertyName: type
      AnotherSampleSchema:
        description: ...
        type: ...
        anotherProperty:
            description: ...
            oneOf:
              - type: string
                enum:
                  - ENDOFBLOCK
                  - ENDOFSURVEY
              - $ref: '#/components/schemas/SurveyID'
        additionalProperties:
          oneOf:
            - $ref: '#/components/schemas/SomeSchema'
            - $ref: '#/components/schemas/SomeOtherSchema'
```
</details>

> To make the compilation to succeed, please remove these keywords (if they exist in the above mentioned positions) from the input specification and generate the client again.
> 
> Please note that while removing these keywords from a property, you must also supply a meaningful replacement.
> Sometimes you may also choose to drop the property altogether for e.g, in the case of `additionalProperties`
> 
> Refer to the [Swagger documentation](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/) to understand more about the usage of these keywords and to derive their replacement.

Reviewer: ```suggestion
To make the compilation succeed, please remove these keywords from your specification and generate the client again.

Please note that while removing these keywords from a property, you must also supply a meaningful replacement.
Sometimes you may also choose to drop the property altogether for e.g, in the case of `additionalProperties`

Refer to the [Swagger documentation](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/) to understand more about the usage of these keywords and to derive their replacement.
```
